presto-main/src/test/java/com/facebook/presto/type/TestDateTimeOperators.java

```diff
@@ -71,4 +84,89 @@ public void testDaylightTimeSavingSwitchCrossingIsNotApplied()
         assertFunction("TIMESTAMP '2013-10-27 03:05' - INTERVAL '1' hour", TIMESTAMP, sqlTimestampOf(2013, 10, 27, 2, 5, 0, 0, TIME_ZONE, TIME_ZONE_KEY, getSession()));
         assertFunction("TIMESTAMP '2013-10-27 03:05' - INTERVAL '2' hour", TIMESTAMP, sqlTimestampOf(2013, 10, 27, 1, 5, 0, 0, TIME_ZONE, TIME_ZONE_KEY, getSession()));
     }
+
+    private static String valueFromLiteral(String literal)
+    {
+        Pattern p = Pattern.compile("'(.*)'");
+        Matcher m = p.matcher(literal);
+        m.find();
+        return m.group(1);
+    }
+
+    private void testTimeRepresentationOnDate(DateTime date, String timeLiteral, Type expectedType, Object expected)
+    {
+        Session localSession = testSessionBuilder()
+                .setTimeZoneKey(getTimeZoneKey("America/Los_Angeles"))
+                .setStartTime(date.getMillis())
+                .setSystemProperty("legacy_timestamp", "false")
+                .build();
+
+        FunctionAssertions localAssertions = new FunctionAssertions(localSession);
+        localAssertions.assertFunction(timeLiteral, expectedType, expected);
+        localAssertions.assertFunctionString(timeLiteral, expectedType, valueFromLiteral(timeLiteral));
+    }
+
+    @Test
+    public void testTimeWithTimeZoneRepresentation()
+    {
+        // PST -> PDT date
+        testTimeRepresentationOnDate(new DateTime(2017, 3, 12, 10, 0, 0, 0, DateTimeZone.UTC),
```

Comment:
First argument on its own line. This comments applies to multiple places in this commit.